### PR TITLE
phpunit 4 parts removed

### PR DESCRIPTION
**Pull request by kokspflanze**
removed old PHPUnit4 Parts
changed `setExpectedException` to `expectException` and `expectExceptionMessage`
added in `testCanCreateServiceWithAbstractFactory` a assert
added PHPUnit6, thanks to @webimpress for the hint=)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "require-dev": {
         "ocramius/proxy-manager": "^1.0 || ^2.0",
         "phpbench/phpbench": "^0.10.0",
-        "phpunit/phpunit": "^4.6 || ^5.2.10",
+        "phpunit/phpunit": "^5.2.10",
```
Comment (webimpress) [inline composer.json]: I think we can use here `^5.7 || ^6.0`

File: test/AbstractFactory/ReflectionBasedAbstractFactoryTest.php [hunk 2]
```diff
@@ -54,8 +54,8 @@ public function testFactoryRaisesExceptionWhenUnableToResolveATypeHintedService(
         $this->container->has(TestAsset\SampleInterface::class)->willReturn(false);
         $this->container->has('config')->willReturn(false);
         $factory = new ReflectionBasedAbstractFactory();
-        $this->setExpectedException(
-            ServiceNotFoundException::class,
+        $this->expectException(ServiceNotFoundException::class);
+        $this->expectExceptionMessage(
```
Comment (webimpress) [inline test/AbstractFactory/ReflectionBasedAbstractFactoryTest.php]: You can move `sprintf` into this line, usually we do like that:

```php
$this->expectExceptionMessage(sprintf(
    'Unable to create service "%s"; unable to resolve parameter "sample" using type hint "%s"',
    TestAsset\ClassWithTypeHintedConstructorParameter::class,
    TestAsset\SampleInterface::class
));
```

File: test/AbstractFactory/ReflectionBasedAbstractFactoryTest.php [hunk 3]
```diff
@@ -68,8 +68,8 @@ public function testFactoryRaisesExceptionWhenUnableToResolveATypeHintedService(
     public function testFactoryRaisesExceptionForScalarParameters()
     {
         $factory = new ReflectionBasedAbstractFactory();
-        $this->setExpectedException(
-            ServiceNotFoundException::class,
+        $this->expectException(ServiceNotFoundException::class);
+        $this->expectExceptionMessage(
```
Comment (webimpress) [inline test/AbstractFactory/ReflectionBasedAbstractFactoryTest.php]: As above, move `sprintf` in the same line.

File: test/AbstractPluginManagerTest.php [hunk 4]
```diff
@@ -181,7 +181,7 @@ function ($container, $name, $callback) {
     public function testGetRaisesExceptionWhenNoFactoryIsResolved()
     {
         $pluginManager = $this->createContainer();
-        $this->setExpectedException(ServiceNotFoundException::class, get_class($pluginManager));
+        $this->expectException(ServiceNotFoundException::class, get_class($pluginManager));
```
Comment (webimpress) [inline test/AbstractPluginManagerTest.php]: `expectException` has only one parameter. Please add `expectExceptionMessage` with `get_class($pluginManager)`.

File: composer.json (hunk 1 above)
Comment (kokspflanze) [inline composer.json]: 5.7 yes, but no reason or?
about 6.0 that will not work, like i wrote in the PR

Comment (webimpress) [inline composer.json]: In PHPUnit 5.7 we already have `PHPUnit\Framework\TestCase` so basically you need replace:
`use PHPUnit_Framework_TestCase as TestCase` with `use PHPUnit\Framework\TestCase` and all should be fine also on PHPUnit 6.0

Comment (kokspflanze) [inline composer.json]: ahh your right, didnt saw that

File: test/AbstractFactory/ConfigAbstractFactoryTest.php [hunk 5]
```diff
@@ -15,7 +15,7 @@
 use ZendTest\ServiceManager\TestAsset\SecondComplexDependencyObject;
 use ZendTest\ServiceManager\TestAsset\SimpleDependencyObject;
 
-class ConfigAbstractFactoryTest extends \PHPUnit_Framework_TestCase
+class ConfigAbstractFactoryTest extends \PHPUnit\Framework\TestCase
```
Comment (webimpress) [inline test/AbstractFactory/ConfigAbstractFactoryTest.php]: Please import `PHPUnit\Framework\TestCase` so then here will be just `TestCase`.

File: test/AbstractFactory/ReflectionBasedAbstractFactoryTest.php [hunk 6]
```diff
@@ -8,7 +8,7 @@
 namespace ZendTest\ServiceManager\AbstractFactory;
 
 use Interop\Container\ContainerInterface;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
Comment (webimpress) [inline test/AbstractFactory/ReflectionBasedAbstractFactoryTest.php]: ` as TestCase` is redundant now and can be removed (everywhere please)

File: test/Tool/FactoryCreatorTest.php [hunk 7]
```diff
@@ -12,7 +12,7 @@
 use ZendTest\ServiceManager\TestAsset\InvokableObject;
 use ZendTest\ServiceManager\TestAsset\SimpleDependencyObject;
 
-class FactoryCreatorTest extends \PHPUnit_Framework_TestCase
+class FactoryCreatorTest extends \PHPUnit\Framework\TestCase
```
Comment (webimpress) [inline test/Tool/FactoryCreatorTest.php]: Please import `PHPUnit\Framework\TestCase`.

Comment (kokspflanze) [inline test/Tool/FactoryCreatorTest.php]: yeah i will check the import parts

**Comment by weierophinney** [comment]
Thanks, @kokspflanze\!

**Pull request closed without merging**